psi4/src/psi4/libmints/electricfield.cc

```diff
@@ -214,6 +214,138 @@ void ElectricFieldInt::compute_pair(const GaussianShell& s1, const GaussianShell
+
+                        P[0] = (a1 * A[0] + a2 * B[0]) * oog;
+                        P[1] = (a1 * A[1] + a2 * B[1]) * oog;
+                        P[2] = (a1 * A[2] + a2 * B[2]) * oog;
+                        PA[0] = P[0] - A[0];
+                        PA[1] = P[1] - A[1];
+                        PA[2] = P[2] - A[2];
+                        PB[0] = P[0] - B[0];
+                        PB[1] = P[1] - B[1];
+                        PB[2] = P[2] - B[2];
+
+                        double over_pf = exp(-a1 * a2 * AB2 * oog) * sqrt(M_PI * oog) * M_PI * oog * c1 * c2;
+
+                        PC[0] = P[0] - C[0];
+                        PC[1] = P[1] - C[1];
+                        PC[2] = P[2] - C[2];
+
+                        // Get recursive
+                        efield_recur_.compute(PA, PB, PC, gamma, am1, am2);
+                        // Gather contributions.
+                        ao12 = 0;
+                        int ao1 = 0;
+                        for (int ii = 0; ii <= am1; ++ii) {
+                            int l1 = am1 - ii;
+                            for (int jj = 0; jj <= ii; ++jj) {
+                                int m1 = ii - jj;
+                                int n1 = jj;
+
+                                int ao2 = 0;
+                                for (int kk = 0; kk <= am2; ++kk) {
+                                    int l2 = am2 - kk;
+                                    for (int ll = 0; ll <= kk; ++ll) {
+                                        int m2 = kk - ll;
+                                        int n2 = ll;
+
+                                        // Compute location in the recursion
+                                        int iind = l1 * ixm + m1 * iym + n1 * izm;
+                                        int jind = l2 * jxm + m2 * jym + n2 * jzm;
+
+                                        buffer_[ao12] += ex[iind][jind][0] * over_pf;
+                                        buffer_[ao12 + ydisp] += ey[iind][jind][0] * over_pf;
+                                        buffer_[ao12 + zdisp] += ez[iind][jind][0] * over_pf;
+                                        functor(ao1 + bf1_offset, ao2 + bf2_offset, site, ex[iind][jind][0] * over_pf,
+                                                ey[iind][jind][0] * over_pf, ez[iind][jind][0] * over_pf);
+                                        ao2++;
+                                        ao12++;
+                                    }
+                                }
+                                ao1++;
+                            }
+                        }
+                    }  // loop over primitives of shell 2
+                }      // loop over primitives of shell 1
+            }          // loop over points
+            bf2_offset += nj;
+        }  // loop over shell 2
+        bf1_offset += ni;
+    }  // loop over shell 1
+}
+
+template void ElectricFieldInt::compute_with_functor(ContractOverDipolesFunctor&, SharedMatrix);
```

Comment:
Shouldn't `template` be followed by `<>` for this full specialization? https://en.cppreference.com/w/cpp/language/template_specialization

Reply:
I thought the same, but then got `Symbol not found` errors when loading the `core` library from Python...

Reply:
Found this: https://en.cppreference.com/w/cpp/language/function_template
The variant I used seems to be OK (case (2) in 'Function template instantiation'), but I really don't know why `template<>` does not work...